include/ctranslate2/primitives/primitives.h

```diff
@@ -197,6 +197,12 @@ namespace ctranslate2 {
                            Out* c);
   };
 
+#ifdef WITH_MKL
+  template<>
+  template<>
+  void primitives<Device::CPU>::mul(float a, float* y, size_t size);
```

Comment:
This overload was indeed not correctly declared. It would be best to avoid using `#ifdef WITH_MKL` in headers. Could you instead make the MKL overload use this signature:

```cpp
void primitives<Device::CPU>::mul(float a, const float* x, float* y, size_t size)
```

and implement it using [cblas_saxpby](https://software.intel.com/en-us/mkl-developer-reference-c-cblas-axpby) (with `b = 0`)?

Reply:
Right.